applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -227,6 +253,12 @@ static void background_erase_handler(struct k_work *work)
 			LOG_ERR("Cannot erase page (%d)", err);
 			flash_area_close(flash_area);
 			flash_area = NULL;
+
+			if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK) &&
+			    !k_work_delayable_is_pending(&dfu_timeout)) {
```

Comment:
That may also break DFU logic since we rely on fact that FLASH area is open to determine that DFU operation is already ongoing

Reply:
Yes, it can. Consider the following scenario:

Config channel DFU source stops sending DFU packages at some point. Once the DFU is still not timed out, the DFU source will restart DFU from scratch setting the image offset to zero. The background erase operation will be scheduled in this corner case before the DFU timeout.

Reply:
Wouldn't the following check in `handle_dfu_start` function prevent the subsequent DFU from starting?
```
        if (flash_area) {                                                                            
                LOG_WRN("DFU already in progress");                                                  
                return;                                                                              
        }
```

Reply:
You are right, but we need to guard against the pending reboot handler